src/MicrosoftAuthFlow.ts

```diff
@@ -25,18 +25,37 @@ async function retry (methodFn, beforeRetry, times) {
   }
 }
 
+type MicrosoftAuthFlowOptions = {
+  flow: 'live' | 'sisu' | 'msal',
+  authTitle?: string | undefined,
+  relyingParty?: string | undefined,
+  password?: string | undefined,
+  fetchEntitlements?: boolean | undefined,
+  fetchProfile?: boolean | undefined,
+  fetchCertificates?: boolean | undefined
+}
+
 class MicrosoftAuthFlow {
-  constructor (username = '', cache = __dirname, options, codeCallback) {
+  username: string
+  options: MicrosoftAuthFlowOptions
+  xbl?: typeof XboxTokenManager | undefined
+  msa?: typeof MsaTokenManager | undefined
+  mba?: typeof BedrockTokenManager | undefined
+  mca?: typeof JavaTokenManager | undefined
+  doTitleAuth?: boolean | undefined
+  codeCallback: (response: any) => void
```

Comment:
The response in this callback can vary depending on if MsaTokenManager is being used or LiveTokenManager.

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/603098e62124b90c13dcd6e57a7d83d95cc07ce8/lib/msal-common/src/response/DeviceCodeResponse.ts#L15

Live - Returns ServerDeviceCodeResponse 
Msa - Returns DeviceCodeResponse